pcap.pyx

```diff
@@ -227,92 +237,85 @@ cdef class pcap:
     property filter:
         """Current packet capture filter."""
         def __get__(self):
-            return self.__filter
-    
+            return self.__filter.decode('UTF-8')
+
     property fd:
         """File descriptor (or Win32 HANDLE) for capture handle."""
         def __get__(self):
             return pcap_ex_fileno(self.__pcap)
-        
+
     def fileno(self):
         """Return file descriptor (or Win32 HANDLE) for capture handle."""
         return pcap_ex_fileno(self.__pcap)
-    
+
     def setfilter(self, value, optimize=1):
         """Set BPF-format packet capture filter."""
         cdef bpf_program fcode
         free(self.__filter)
-        self.__filter = strdup(value)
+        py_byte_value = value.encode('UTF-8')
+        self.__filter = strdup(py_byte_value)
```

Comment:
Since here we are using it as it's byte value, what is the reason for using `unicode` above instead of using str in py2 and bytes in py3?